iroh-util/src/lib.rs

```diff
@@ -44,15 +44,43 @@ pub async fn block_until_sigint() {
     ctrlc_oneshot.await.unwrap();
 }
 
-/// Path to the iroh home directory.
-pub fn iroh_home_root() -> Option<PathBuf> {
-    let home = home_dir()?;
-    Some(Path::new(&home).join(IROH_DIR))
+/// Returns the path to the user's iroh config directory.
+///
+/// The returned value depends on the operating system and is either a `Some`, containing a value from the following table, or a `None`.
+///
+/// | Platform | Value                                 | Example                          |
+/// | -------- | ------------------------------------- | -------------------------------- |
+/// | Linux    | `$XDG_CONFIG_HOME` or `$HOME`/.config/iroh | /home/alice/.config/iroh              |
+/// | macOS    | `$HOME`/Library/Application Support/iroh   | /Users/Alice/Library/Application Support/iroh |
+/// | Windows  | `{FOLDERID_RoamingAppData}`/iroh           | C:\Users\Alice\AppData\Roaming\iroh   |
+pub fn iroh_config_root() -> Option<PathBuf> {
+    let cfg = dirs_next::config_dir()?;
+    Some(cfg.join(&IROH_DIR))
+}
+
+// Path that leads to a file in the iroh config directory.
+pub fn iroh_config_path(file_name: &str) -> Option<PathBuf> {
+    let path = iroh_config_root()?.join(file_name);
+    Some(path)
+}
+
+/// Returns the path to the user's iroh data directory.
+///
+/// The returned value depends on the operating system and is either a `Some`, containing a value from the following table, or a `None`.
+///
+/// | Platform | Value                                         | Example                                  |
+/// | -------- | --------------------------------------------- | ---------------------------------------- |
+/// | Linux    | `$XDG_DATA_HOME`/iroh or `$HOME`/.local/share/iroh | /home/alice/.local/share/iroh                 |
+/// | macOS    | `$HOME`/Library/Application Support/iroh      | /Users/Alice/Library/Application Support/iroh |
+/// | Windows  | `{FOLDERID_RoamingAppData}/iroh`              | C:\Users\Alice\AppData\Roaming\iroh           |
+pub fn iroh_data_root() -> Option<PathBuf> {
```

Comment:
Lets return `Result`s for these, and error when they are not available. All uses cases atm are to error if we do't have a path

Reply:
in2it. it'll make for a more explicit division from embedded systems